src/main/webapp/app/exercises/programming/manage/programming-exercise.component.ts

```diff
@@ -130,12 +105,46 @@ export class ProgrammingExerciseComponent extends ExerciseComponent implements O
 
     openOrionEditor(exercise: ProgrammingExercise) {
         try {
-            this.router.navigate(['code-editor', 'ide', exercise.id, 'admin', exercise.templateParticipation?.id]);
+            this.router.navigate(['code-editor', 'ide', exercise.id, 'admin', exercise.templateParticipation!.id!]);
         } catch (e) {
             this.javaBridge.log(e);
         }
     }
 
+    getHiddenTestCasesNumber(programmingExerciseId: number): number {
+        let number = 0;
+        this.gradingService
+            .subscribeForTestCases(programmingExerciseId)
+            .pipe(
+                tap((testCases: ProgrammingExerciseTestCase[]) => {
+                    testCases.forEach((testCase) => {
+                        if (testCase.afterDueDate) {
+                            number++;
+                        }
+                    });
+                }),
+            )
+            .subscribe();
+        return number;
+    }
+
+    getPublicTestCasesNumber(programmingExerciseId: number): number {
+        let number = 0;
+        this.gradingService
+            .subscribeForTestCases(programmingExerciseId)
+            .pipe(
+                tap((testCases: ProgrammingExerciseTestCase[]) => {
+                    testCases.forEach((testCase) => {
+                        if (!testCase.afterDueDate) {
+                            number++;
+                        }
+                    });
+                }),
+            )
+            .subscribe();
+        return number;
+    }
```

Comment:
I don't think you need two requests here. I would call this once for each programmingEx in onInit and create dict mapping exercise id to those two numbers. Then you can easily access those in the template.